types/three/three-core.d.ts

```diff
@@ -5230,18 +5241,23 @@ export const LinePieces: number;
 export class LineSegments extends Line {
     constructor(
         geometry?: Geometry | BufferGeometry,
-        material?: LineDashedMaterial | LineBasicMaterial | ShaderMaterial | (LineDashedMaterial | LineBasicMaterial | ShaderMaterial)[],
+        material?: LineMaterialType | LineMaterialType[],
         mode?: number
     );
 }
 
-type MeshMaterial = MeshBasicMaterial | MeshLambertMaterial | MeshPhongMaterial | MeshDepthMaterial | MeshStandardMaterial | MeshPhysicalMaterial | MeshNormalMaterial | MeshFaceMaterial | ShaderMaterial;
+/**
+ * An intermediate material type that casts more precisely the possible materials assignable to a [[Mesh]] object.
+ *
+ * `MeshToonMaterial` and [[MeshPhysicalMaterial]] are omitted as they extend [[MeshPhongMaterial]] and [[MeshStandardMaterial]] respectively.
+ */
+export type MeshMaterialType = MeshBasicMaterial | MeshDepthMaterial | MeshFaceMaterial | MeshLambertMaterial | MeshNormalMaterial | MeshPhongMaterial | MeshStandardMaterial | ShaderMaterial | ShadowMaterial;
```

Comment:
`ShadowMaterial` can be removed from this union since it extends from `ShaderMaterial`, right? Also, shouldn't we be using `MultiMaterial` instead of `MeshFaceMaterial` since `MeshFaceMaterial` extends from `MultiMaterial`?

Reply:
Currently in three.js, 
- `ShadowMaterial` directly extends `Material`.
- both `MultiMaterial` and `MeshFaceMaterial` have been replaced by an array of `Material`. I kept it because @types/physijs uses it and that I was too lazy to remove it from everywhere it was still used :).